### PR TITLE
Bug 1171928 use tabzilla static across mozorg

### PR DESCRIPTION
**Pull request by malena**
https://bugzilla.mozilla.org/show_bug.cgi?id=1171928

### CODE REVIEW

File: bedrock/base/templates/base-resp.html [hunk 1]
```diff
@@ -60,6 +56,7 @@
     {# Global styles, hidden from IE7 and lower #}
     {% block site_css %}
       {% stylesheet 'responsive' %}
+      {% stylesheet 'tabzilla-static' %}
```
Comment (alexgibson) [inline bedrock/base/templates/base-resp.html]: I haven't dug into this PR yet, but given that the `tabzilla-static` CSS file is pretty tiny - is it actually a performance hit to include it like this and incur an extra http request, as opposed to bundling it in the `responsive` common CSS bundle?

Comment (malena) [inline bedrock/base/templates/base-resp.html]: Yes the only css in the 'responsive' bundle is 'sandstone-resp.less', so I can definitely add it. Glad you suggested. I caught a few other places where tabzilla-static was part of page specific bundles I can remove.

File: bedrock/foundation/templates/foundation/annualreport/2011.html [hunk 2]
```diff
@@ -11,6 +11,7 @@
 {% block body_class %}sand{% endblock %}
 
 {% block extrahead %}
+  {% stylesheet 'tabzilla-static' %}
```
Comment (jpetto) [inline bedrock/foundation/templates/foundation/annualreport/2011.html]: Could we add `tabzilla-static.less` to the `annual_2011` bundle instead of including two separate stylesheets?

Comment (malena) [inline bedrock/foundation/templates/foundation/annualreport/2011.html]: Ah yes! Will definitely change that.

File: media/css/firefox/desktop/trust.less [hunk 3]
```diff
@@ -4,9 +4,10 @@
 
 @import 'common.less';
 
-// customize family nav
-.fxfamilynav-tabzilla:before {
-    background-color: @textColorSecondary;
+#fxfamilynav-tabzilla {
+    &:before {
```
Comment (sgarrity) [inline media/css/firefox/desktop/trust.less]: Could these nested LESS selectors be simplified? To this:
```
#fxfamilynav-tabzilla:before {
```

File: bedrock/firefox/templates/firefox/all.html [hunk 4]
```diff
@@ -30,7 +30,7 @@
   sky firefox-all-{{ channel }} {% if channel == 'alpha' %}blueprint{% endif %}
 {% endblock %}
 
-{% block site_css %}
+{% block page_css %}
```
Comment (jpetto) [inline bedrock/firefox/templates/firefox/all.html]: The `firefox_all` bundle in `static_media.py` still references `sandstone-resp.less`, which results in duplicate CSS.

File: bedrock/firefox/templates/firefox/desktop/index.html [hunk 5]
```diff
@@ -33,9 +33,6 @@
   {% endcall %}
 {% endblock %}
 
-{% block tabzilla_css %}{% endblock %}
-{% block tabzilla_js %}{% endblock %}
```
Comment (jpetto) [inline bedrock/firefox/templates/firefox/desktop/index.html]: These empty blocks still exist in many templates. Should remove everywhere, no?

File: bedrock/firefox/templates/firefox/all.html (hunk 4 above)
Comment (jpetto) [inline bedrock/firefox/templates/firefox/all.html]: On a related note, moving all these pages from `site_css` to `page_css` will result in 1 extra HTTP request per page, as we're now loading two separate bundles. Are we okay adding this overhead?

**Comment by jpetto** [comment]
In review, I noticed at least a couple pages (`/firefox/pocket/` and `/firefox/independent/`) that load `sandstone-resp.less` twice - once in the `page_css` bundle and once in the `site_css` bundle. This PR did not introduce the code duplication, but is touching those CSS blocks, so I figured I'd mention it here. Should be cleaned up sooner rather than later, whether here or in a separate PR.

File: bedrock/firefox/templates/firefox/all.html (hunk 4 above)
Comment (craigcook) [inline bedrock/firefox/templates/firefox/all.html]: > On a related note, moving all these pages from site_css to page_css will result in 1 extra HTTP request per page, as we're now loading two separate bundles. Are we okay adding this overhead?

I think we need to weigh the potential cost of the extra request against the potential savings of smaller files, especially if the global site styles are cached from a previous page. I'd love to do some performance testing to compare load times.

File: media/css/firefox/family-nav.less [hunk 6]
```diff
@@ -4,6 +4,12 @@
 
 @import "../sandstone/lib.less";
 
+#tabzilla {
+    &:before {
+        background-color: @textColorSecondary;
+    }
+}
+
```
Comment (jpetto) [inline media/css/firefox/family-nav.less]: This selector isn't necessary as the file stands. The family nav has its own tabzilla implementation (down on line 57). However, it probably is best to switch the family nav over to the standard implementation...which would require a little bit more work. Could be broken out to a separate PR - up to you.

File: bedrock/settings/static_media.py [hunk 7]
```diff
@@ -826,6 +807,12 @@
         ),
         'output_filename': 'css/tabzilla-min.css',
     },
+    'tabzilla-static': {
```
Comment (jpetto) [inline bedrock/settings/static_media.py]: Doesn't look like this is used anymore after the change to the 2011 annual report?

File: bedrock/firefox/templates/firefox/all.html (hunk 4 above)
Comment (jgmize) [inline bedrock/firefox/templates/firefox/all.html]: >> On a related note, moving all these pages from site_css to page_css will result in 1 extra HTTP request per page, as we're now loading two separate bundles. Are we okay adding this overhead?

> I think we need to weigh the potential cost of the extra request against the potential savings of smaller files, especially if the global site styles are cached from a previous page. I'd love to do some performance testing to compare load times.

+1, I've been guilty of [obsessing over HTTP request reduction](http://blog.getify.com/obsessions-http-request-reduction/) myself, but sometimes splitting the files up can yield better performance, despite the request overhead.

File: bedrock/firefox/templates/firefox/desktop/index.html (hunk 5 above)
Comment (malena) [inline bedrock/firefox/templates/firefox/desktop/index.html]: Yes!

File: bedrock/firefox/templates/firefox/all.html (hunk 4 above)
Comment (jpetto) [inline bedrock/firefox/templates/firefox/all.html]: Yeah, seems from a macro level to be an "it depends" situation. Does a cached resource count against the max parallel HTTP requests limit? The home page currently has 30 HTTP requests, and it seems the max parallel limit is around 8 in modern browsers, so maybe it doesn't matter? This is outside of my ken. :smile_cat:

Bring it up in tomorrow's meeting?

Comment (jpetto) [inline bedrock/firefox/templates/firefox/all.html]: FWIW, my gut feels the approach here (shared/cached bundle and page-specific bundle) is the right thing to do, but my gut cannot produce any supporting evidence.

**Comment by jpetto** [comment]
Can remove a [little bit of the instructions](https://github.com/mozilla/bedrock/blob/master/bedrock/base/templates/macros.html#L503) for the family nav macro.

**Comment by alexgibson** [comment]
Friendly drive-by comment. If you rebase, there's a couple more empty tabzilla blocks that can be removed here when this lands:

https://github.com/mozilla/bedrock/blob/master/bedrock/firefox/templates/firefox/hello/start-40.0.html#L62-L64

:car: :runner:

File: bedrock/firefox/templates/firefox/all.html (hunk 4 above)
Comment (malena) [inline bedrock/firefox/templates/firefox/all.html]: As discussed in meeting I'm going to leave the page_css block in. We can make another bug for testing better performance.

**Comment by malena** [comment]
Thanks @alexgibson. Removing.

**Comment by malena** [comment]
I've tackled the commented issues. Another review would be great!

**Comment by malena** [comment]
@jpetto Whenever you have a chance this is ready to be reviewed.

**Comment by jpetto** [comment]
Wondering if we should update the [tabzilla docs](http://bedrock.readthedocs.org/en/latest/tabzilla.html) as well. The docs have instructions for non-bedrock sites, so maybe not yet, but should probably file a bug to keep it on the radar?

File: bedrock/firefox/templates/firefox/australis/fx38_0_5/firstrun.html [hunk 8]
```diff
@@ -23,10 +23,7 @@
 
 {% block body_class %}mozID{% endblock %}
 
-{% block tabzilla_css %}{% endblock %}
-{% block tabzilla_js %}{% endblock %}
-
-{% block site_css %}
+{% block page_css %}
   {% stylesheet 'firefox_fx38_0_5_firstrun' %}
```
Comment (jpetto) [inline bedrock/firefox/templates/firefox/australis/fx38_0_5/firstrun.html]: This bundle needs pruning - has its own reference to `sandstone.less` and `tabzilla-static.less`.

File: bedrock/firefox/templates/firefox/australis/tour-base.html [hunk 9]
```diff
@@ -23,8 +23,6 @@
 
 {% block body_class %}sky{% endblock %}
 
-{% block site_css %}{% endblock %}
```
Comment (jpetto) [inline bedrock/firefox/templates/firefox/australis/tour-base.html]: Pages using this base template (e.g. [firefox/australis/firstrun-tour.html](https://github.com/malena/bedrock/blob/bug-1171928-use-tabzilla-static-across-mozorg/bedrock/firefox/templates/firefox/australis/firstrun-tour.html#L27)) end up with flip-flopped CSS order. The page specific stuff ends up in `site_css`, while the common global stuff ends up in `page_css` (due to inheriting from [firefox/base.html](https://github.com/malena/bedrock/blob/bug-1171928-use-tabzilla-static-across-mozorg/bedrock/firefox/templates/firefox/base.html#L18)).

It's kind of a goose chase to follow the inheritance chain. :confused: However, you can rather easily hack the site to load a specific firstrun template by adding a line like `template = 'firefox/australis/firstrun-tour.html'` to [line 493 in `firefox/views.py`](https://github.com/malena/bedrock/blob/bug-1171928-use-tabzilla-static-across-mozorg/bedrock/firefox/views.py#L493).

File: bedrock/firefox/templates/firefox/base-resp.html [hunk 10]
```diff
@@ -18,7 +18,7 @@
 {% block twitter_id %}firefox{% endblock %}
 
 {% block site_css %}
-  {% stylesheet 'firefox-resp' %}
+    {% stylesheet 'firefox-resp' %}
```
Comment (jpetto) [inline bedrock/firefox/templates/firefox/base-resp.html]: Nit - extra indentation (only because you're still making updates :smile:)

File: bedrock/firefox/templates/firefox/base.html [hunk 11]
```diff
@@ -15,7 +15,7 @@
 {% block facebook_id %}14696440021{# facebook.com/Firefox #}{% endblock %}
 {% block twitter_id %}firefox{% endblock %}
 
-{% block site_css %}
+{% block page_css %}
```
Comment (jpetto) [inline bedrock/firefox/templates/firefox/base.html]: Hm, in `firefox/base-resp.html`, you leave this block as `site_css`. Any reason this template should be different?

File: bedrock/firefox/templates/firefox/dashboard-base.html [hunk 12]
```diff
@@ -7,7 +7,7 @@
 {% block page_title_prefix %}{% endblock %}
 {% block page_title %}{% endblock %}
 
-{% block site_css %}
+{% block page_css %}
   {% stylesheet 'firefox-dashboard' %}
```
Comment (jpetto) [inline bedrock/firefox/templates/firefox/dashboard-base.html]: Just noticed this bundle loads `menu-resp.less`, which isn't needed on the two templates that inherit from this template (`firefox/interest-dashboard.html` and `firefox/tiles.html`).

Also, those two inheriting pages have their CSS in `site_css`, which results in a weird CSS load order.

File: bedrock/firefox/templates/firefox/desktop/index.html [hunk 13]
```diff
@@ -21,7 +21,7 @@
   {% endif %}
 {% endblock %}
 
-{% block site_css %}
+{% block page_css %}
```
Comment (jpetto) [inline bedrock/firefox/templates/firefox/desktop/index.html]: The other desktop pages (`fast.html`, `trust.html`, `customize.html`) still use the `site_css` block. Should those be switch to `page_css` as well?

File: bedrock/firefox/templates/firefox/family/index.html [hunk 14]
```diff
@@ -25,9 +25,6 @@
   {% endif %}
 {% endblock %}
 
-{% block tabzilla_css %}{% endblock %}
-{% block tabzilla_js %}{% endblock %}
-
```
Comment (jpetto) [inline bedrock/firefox/templates/firefox/family/index.html]: This page is still using the `site_css` block. Should we switch to `page_css`?

File: bedrock/firefox/templates/firefox/hello/index.html [hunk 15]
```diff
@@ -52,9 +52,6 @@
 data-try-now="{{ _('Try it now') }}"
 {% endblock %}
 
-{% block tabzilla_css %}{% endblock %}
-{% block tabzilla_js %}{% endblock %}
-
```
Comment (jpetto) [inline bedrock/firefox/templates/firefox/hello/index.html]: This page is still using the `site_css` block as well.

File: bedrock/firefox/templates/firefox/interest-dashboard.html [hunk 16]
```diff
@@ -24,9 +24,6 @@
   {{ fxfamilynav('desktop', '', 'dark') }}
 {% endblock %}
 
-{% block tabzilla_css %}{% endblock %}
-{% block tabzilla_js %}{% endblock %}
-
```
Comment (jpetto) [inline bedrock/firefox/templates/firefox/interest-dashboard.html]: Another one using `site_css`.

File: bedrock/firefox/templates/firefox/tiles.html [hunk 17]
```diff
@@ -31,9 +31,6 @@
   {{ fxfamilynav('desktop', '', 'dark') }}
 {% endblock %}
 
-{% block tabzilla_css %}{% endblock %}
-{% block tabzilla_js %}{% endblock %}
-
```
Comment (jpetto) [inline bedrock/firefox/templates/firefox/tiles.html]: This page has confusing CSS inheritance. Might be better to switch this template from `site_css` to `page_css`?

File: media/css/firefox/desktop/trust.less [hunk 18]
```diff
@@ -173,6 +172,11 @@ html[dir="rtl"] {
 }
 
 @media (max-width: @breakTablet) {
+
+    #fxfamilynav-tabzilla:before {
+        background-color: transparent; 
+    }
```
Comment (jpetto) [inline media/css/firefox/desktop/trust.less]: Nice catch!

File: media/css/firefox/family-nav.less [hunk 19]
```diff
@@ -4,6 +4,10 @@
 
 @import "../sandstone/lib.less";
 
+#tabzilla:before {
+    background-color: @textColorSecondary;
+}
```
Comment (jpetto) [inline media/css/firefox/family-nav.less]: Is this selector necessary? Most (all?) pages using the family nav do not have an element with the id `tabzilla` - instead `fxfamilynav-tabzilla` is used.

File: media/css/styleguide/styleguide.less [hunk 20]
```diff
@@ -36,6 +36,10 @@ a[href="#TODO"] { color: #f00 !important; }
   .offset(3);
 }
 
+#tabzilla:before {
+    background-color: #484848;
```
Comment (jpetto) [inline media/css/styleguide/styleguide.less]: Use `@textColorSecondary` instead?

**Comment by jpetto** [comment]
A little bit more cleanup to do. Almost there! :persevere:

File: bedrock/firefox/templates/firefox/base.html (hunk 11 above)
Comment (malena) [inline bedrock/firefox/templates/firefox/base.html]: Probably my mistake when updating site_css to page_css. This is a base template so it should stay site_css. Fixing this.

File: bedrock/firefox/templates/firefox/base-resp.html (hunk 10 above)
Comment (malena) [inline bedrock/firefox/templates/firefox/base-resp.html]: np :)

File: bedrock/firefox/templates/firefox/dashboard-base.html (hunk 12 above)
Comment (malena) [inline bedrock/firefox/templates/firefox/dashboard-base.html]: Thinking the dashboard-base.html shouldn't change to page_css instead leave as an overwriting site_css. Otherwise the CSS load order gets wacky since the child templates have their own site overwrites like below in tiles.html

{% block site_css %}
  {{ super() }}
  {% stylesheet 'firefox-tiles' %}
{% endblock %}

Perhaps leaving interest-dashboard.html with a site_css block and the two child templates can change to the page_css block.

What do you think @jpetto?

Not really sure what this {{ super() }} does but if it doesn't affect if I use page_css or site_css block, I would think using page_css would be better for the child templates.

File: bedrock/firefox/templates/firefox/desktop/index.html (hunk 13 above)
Comment (malena) [inline bedrock/firefox/templates/firefox/desktop/index.html]: Reason they remain site_css is because they don't need tabzilla-static css because these templates all use the familynav. However index.html should also remain as site_css otherwise it has extra css it doesn't need. I would need to add back sandstone-less file into the  firefox-desktop bundle, since thats the only css index.html needed to inherit.

For me to update the site_css to page_css for these I would need to update the base template to use familynav. But since we don't want that (I am assuming familynav was specifically added to pages) for now leaving them as site_css and making a PR for maybe down the road removing this familynav implemenation of the tabzilla tab and using tabzilla-static instead. 

The only thing I'll be fixing here is adding back sandstone-less to firefox-desktop and updating the index to site_css block since firefox-desktop would have all the needed css.

Does that make sense @jpetto?

File: bedrock/firefox/templates/firefox/family/index.html (hunk 14 above)
Comment (malena) [inline bedrock/firefox/templates/firefox/family/index.html]: Yes I can switch to page_css and there is no issue besides the fact that it inherits template-resp.css as well. Doesn't change the layout, but wondering if we even need template-resp.css at all? This may be another PR. 

Should we update or leave as is? @jpetto. I'm tempted to update, even if it adds this template-resp.css.

Comment (malena) [inline bedrock/firefox/templates/firefox/family/index.html]: Thinking I will create a new bug to tackle removing template-resp.css @alexgibson, @jpetto. For now I will update to page_css.

**Comment by jbertsch** [comment]
Should we use the new tabzilla that espressive built here?  https://github.com/mozilla/tabzilla

File: bedrock/firefox/templates/firefox/hello/index.html (hunk 15 above)
Comment (malena) [inline bedrock/firefox/templates/firefox/hello/index.html]: This page uses the familynav not tabzilla which is why it uses site_css to overwrite. Will leave as is and we can tackle replacing familynav implementation with tabzilla in another PR.

File: bedrock/firefox/templates/firefox/interest-dashboard.html (hunk 16 above)
Comment (malena) [inline bedrock/firefox/templates/firefox/interest-dashboard.html]: This page also uses familynav not tabzilla. Changing it to page_css breaks the menu. Thinking a few PR's need to come out of the bug, one of them being replacing the familynav with tabzilla-static.

**Comment by malena** [comment]
@jbertsch Yep! I was pointed to this doc http://bedrock.readthedocs.org/en/latest/tabzilla.html#tabzilla-static, but I believe it's the same tabzilla-static. :)  

However there are a few sections with a different implementation of the tabzilla called familynav. I'm thinking it would be best to create another PR to replace those. It would be best to get this PR merged first. 

Thoughts @alexgibson @jpetto @craigcook?

File: media/css/styleguide/styleguide.less (hunk 20 above)
Comment (malena) [inline media/css/styleguide/styleguide.less]: Yep, updating to var.

**Comment by jpetto** [comment]
Agree that the Fx family nav tabzilla update should happen in a separate PR.

File: bedrock/settings/static_media.py [hunk 21]
```diff
@@ -349,8 +343,6 @@
     },
     'firefox_fx38_0_5_firstrun': {
         'source_filenames': (
-            'css/sandstone/sandstone.less',
-            'css/tabzilla/tabzilla-static.less',
```
Comment (jpetto) [inline bedrock/settings/static_media.py]: Gah. So now this page no longer has `tabzilla-static.less`, because it's not included in the `firefox` bundle (which this page inherits from `firefox/base.html`). I'll leave this one up to you - should `tabzilla-static.less` be included in the `firefox` bundle, or added a la carte in each page?

File: bedrock/firefox/templates/firefox/australis/tour-base.html (hunk 9 above)
Comment (jpetto) [inline bedrock/firefox/templates/firefox/australis/tour-base.html]: So the `firefox/australis/firstrun-tour.html` page has broken/missing CSS, but I think only because it's using the `site_css` block. Testing locally, switching `firefox/australis/firstrun-tour.html` to use `page_css` instead fixes the issue.

Comment (jpetto) [inline bedrock/firefox/templates/firefox/australis/tour-base.html]: Ah, also, `firefox/australis/firstrun-tour.html` doesn't load `tabzilla-static.less`. Maybe another reason to add it to the `firefox` LESS bundle?

File: bedrock/firefox/templates/firefox/dashboard-base.html (hunk 12 above)
Comment (jpetto) [inline bedrock/firefox/templates/firefox/dashboard-base.html]: Yep, I think leaving `site_css` in `dashboard-base.html` and using `page_css` in `interest-dashboard.html` and `tiles.html` is the way to go.

`{{ super() }}` basically means "insert the content of the parent template's block", meaning you can kind of extend a block instead of straight up overwriting.

So, I think `dashboard-base.html` would look like:

```
{% block site_css %}
  {# brings in the site_css bundle specified in the parent template (firefox/base-resp.html) #}
  {{ super() }}
  {% stylesheet 'firefox-dashboard' %}
{% endblock %}
```

Each child page would then have a basic `page_css` block, e.g.

```
{% block page_css %}
  {% stylesheet 'firefox-interest-dashboard' %}
{% endblock %}
```

File: bedrock/firefox/templates/firefox/family/index.html (hunk 14 above)
Comment (jpetto) [inline bedrock/firefox/templates/firefox/family/index.html]: Agree that updating to `page_css` is the way to go for this PR.

I opened [bug 1186605](https://bugzilla.mozilla.org/show_bug.cgi?id=1186605) a couple days ago to audit LESS & JS bundles. Please make a note of this instance in that bug so we don't forget. :smile:

File: bedrock/firefox/templates/firefox/hello/index.html (hunk 15 above)
Comment (jpetto) [inline bedrock/firefox/templates/firefox/hello/index.html]: I'm okay leaving this page using `site_css`, as long as we make a note that it needs to be updated when we update tabzilla in the Fx family nav.

Do we have a bug for that yet? If not, can you file one and note this page will need its CSS block/bundle updated?

File: bedrock/firefox/templates/firefox/interest-dashboard.html (hunk 16 above)
Comment (jpetto) [inline bedrock/firefox/templates/firefox/interest-dashboard.html]: As noted above, I think we can change this block to `page_css` as long as `dashboard-base.html` gets updated as well.

File: bedrock/firefox/templates/firefox/tiles.html (hunk 17 above)
Comment (jpetto) [inline bedrock/firefox/templates/firefox/tiles.html]: As noted above, as long as we change `dashboard-base.html` to `site_css`, let's change this page to `page_css`.

File: bedrock/settings/static_media.py (hunk 21 above)
Comment (jpetto) [inline bedrock/settings/static_media.py]: Feel free to do whatever is easier here (which may be just adding `tabzilla-static.less` to this page-specific bundle). I don't want to make us add any more high level decisions to this already large PR. We can re-think when we tackle [bug 1186605](https://bugzilla.mozilla.org/show_bug.cgi?id=1186605) if necessary.

**Comment by jpetto** [comment]
I think we're just about there! This is hopefully the last round of feedback. :construction_worker:

File: bedrock/firefox/templates/firefox/hello/index.html (hunk 15 above)
Comment (malena) [inline bedrock/firefox/templates/firefox/hello/index.html]: I've opened a bug for this issue. https://bugzilla.mozilla.org/show_bug.cgi?id=1188852

File: bedrock/firefox/templates/firefox/family/index.html (hunk 14 above)
Comment (malena) [inline bedrock/firefox/templates/firefox/family/index.html]: Great @jpetto! I have added another note in bug 1186605 regarding possible removal of template-resp.css.

File: bedrock/firefox/templates/firefox/dashboard-base.html (hunk 12 above)
Comment (malena) [inline bedrock/firefox/templates/firefox/dashboard-base.html]: @jpetto Thanks for explanation. I've also seen 

{% block extrahead %}
  {% stylesheet 'foo' %}
{% endblock %}

Does this do the same? Just asking to understand. Either way I've updated to the changes you have noted above. So now child templates extending from dashboard-base.html will have a page_css block instead.

File: bedrock/firefox/templates/firefox/tiles.html (hunk 17 above)
Comment (malena) [inline bedrock/firefox/templates/firefox/tiles.html]: Yup! Changes made.

File: bedrock/firefox/templates/firefox/dashboard-base.html (hunk 12 above)
Comment (jpetto) [inline bedrock/firefox/templates/firefox/dashboard-base.html]: So `{% block extrahead %}` is basically a general purpose container for anything you might want to put in the `<head>` - CSS, JS, `<meta>` tags, etc. If you place CSS bundles in it, they will be shown to *all* browsers, whereas CSS bundles placed in `site_css` and `page_css` will be hidden from IE 7 and below. In short, it's a way to force CSS on IE 7 and lower.

File: bedrock/firefox/templates/firefox/interest-dashboard.html (hunk 16 above)
Comment (malena) [inline bedrock/firefox/templates/firefox/interest-dashboard.html]: yep! updated dashboard-base.html

File: bedrock/settings/static_media.py (hunk 21 above)
Comment (malena) [inline bedrock/settings/static_media.py]: I added tabzilla-static.less back to the firefox bundle because this is a shared bundle as referenced within firefox/base.html.

It wasn't inheriting styles correctly because the base was including the firefox bundle inside of the page_css block instead of the site_css block. 

So basically all base.html templates should not be using the page_css block otherwise that would most likely be overwritten by child pages (as it should be).

File: bedrock/firefox/templates/firefox/australis/fx38_0_5/firstrun.html (hunk 8 above)
Comment (malena) [inline bedrock/firefox/templates/firefox/australis/fx38_0_5/firstrun.html]: removed sandston.less and tabzilla.less from firefox_fx38_0_5_firstrun bundle

File: bedrock/firefox/templates/firefox/australis/tour-base.html (hunk 9 above)
Comment (malena) [inline bedrock/firefox/templates/firefox/australis/tour-base.html]: Added tabzilla and sandstone to firefox less bundle which should be inherited by all the australis templates.

File: media/css/firefox/family-nav.less (hunk 19 above)
Comment (malena) [inline media/css/firefox/family-nav.less]: Yeah I think this was my first edits when I missed some dependencies. I removed it.

**Comment by malena** [comment]
@jpetto I've tackled the rest of the commented issues. Ready for another review.

File: bedrock/firefox/templates/firefox/desktop/index.html (hunk 13 above)
Comment (jpetto) [inline bedrock/firefox/templates/firefox/desktop/index.html]: Yep, this makes sense. We'll clean it up in [bug 1186605](https://bugzilla.mozilla.org/show_bug.cgi?id=1186605).

File: bedrock/firefox/templates/firefox/desktop/tips.html [hunk 22]
```diff
@@ -12,15 +12,16 @@
...
   {% stylesheet 'firefox_desktop_tips' %}
 {% endblock %}
 
 {% block site_header %}
   <header id="masthead">
     <div class="container">
-      <a href="{{ url('mozorg.home') }}" id="tabzilla" data-infobar="{{ settings.TABZILLA_INFOBAR_OPTIONS }}">Mozilla</a>
-
+      <div id="tabzilla">
+        <a href="{{ url('mozorg.home') }}" data-infobar="{{ settings.TABZILLA_INFOBAR_OPTIONS }}">Mozilla</a>
```
Comment (jpetto) [inline bedrock/firefox/templates/firefox/desktop/tips.html]: We no longer need the `data-infobar` attribute here, as static tabzilla doesn't support any additional infobar messaging.

File: bedrock/foundation/templates/foundation/annualreport/2011.html [hunk 23]
```diff
@@ -25,7 +25,9 @@
 
 {% block site_header %}
   <header id="masthead">
-    <a href="{{ url('mozorg.home') }}" id="tabzilla" data-infobar="{{ settings.TABZILLA_INFOBAR_OPTIONS }}">{{ _('Mozilla') }}</a>
+    <div id="tabzilla">
+      <a href="{{ url('mozorg.home') }}" data-infobar="{{ settings.TABZILLA_INFOBAR_OPTIONS }}">{{ _('Mozilla') }}</a>
```
Comment (jpetto) [inline bedrock/foundation/templates/foundation/annualreport/2011.html]: Don't need the `data-infobar` attribute here or below.

File: media/css/styleguide/styleguide.less (hunk 20 above)
Comment (jpetto) [inline media/css/styleguide/styleguide.less]: Forgot to update to `@textColorSecondary`?

Comment (malena) [inline media/css/styleguide/styleguide.less]: ooops.. must have edited /static instead of /media. Added back in.

**Comment by malena** [comment]
@jpetto Removed data-infobar in tabzilla and updated less. Good for final review hopefully!!

**Comment by jpetto** [comment]
Okay, for the final check, I've been giving the site a visual once over. Found a few issues:

- `/firefox/releases/` is missing `tabzilla-static.less`. Looks like this template (`/firefox/releases/index.html`) needs updating to `page_css`, and also needs its `firefox_releases_index` LESS bundle updated.
- `/firefox/new/` shouldn't rely on the `site_css` block. This page should use the `extrahead` block to make sure *all* CSS is sent to *all* browsers (`site_css` and `page_css` are hidden from IE 7 and below). Currently, this page is a little off in terms of CSS in IE 7. Let's revert this page's LESS bundle, then add `tabzilla-static.less` to it. `/firefox/new/` is in dire need of improvements, which should be starting soon in [bug 1187057](https://bugzilla.mozilla.org/show_bug.cgi?id=1187057).
- Styleguide sub-pages have ended up with a duplicate next/prev nav kind of tucked away under tabzilla:
screen shot 2015-07-30 at 12 51 17 pm

So...close... :partly_sunny:

**Comment by malena** [comment]
Hey @jpetto I've updated the first and third bullet point. 

For the second bullet, the firefox/new page is looking fine to me. The firefox/new.html page is currently using the block extrahead: https://github.com/malena/bedrock/blob/bug-1171928-use-tabzilla-static-across-mozorg/bedrock/firefox/templates/firefox/new.html#L37 and it's also inheriting the tabzilla-static. No need to add it back in to the firefox_new bundle?

Maybe I'm missing something from this second point?

**Comment by jpetto** [comment]
The problem with `/firefox/new/` only shows up in IE 7 or below. Those browsers do not get any CSS loaded in `site_css` or `page_css` - meaning no `tabzilla-static.less` inheritance. Here's how it looks in my IE 7 VM:

screen shot 2015-07-30 at 3 46 46 pm

This is kind of the reason the `extrahead` block exists - for the outlier pages that can't follow our standard CSS degradation rules.

`/firefox/new/` is probably the most important page on the site, which is why we want it to look nice, even on *super* old versions of IE. In the future, this page will likely become even more of a special & unique :snowflake:, possibly not inheriting from any base templates at all.

The styleguide and releases pages look good. One more to go!

**Comment by malena** [comment]
Sorry @jpetto. Totally end of day brain fart. K it's all updated.

File: bedrock/firefox/templates/firefox/new.html [hunk 24]
```diff
@@ -33,7 +33,6 @@
   <h2>{{ high_res_img('img/firefox/new/header-firefox.png', {'alt': _('Firefox for desktop'), 'width': '223', 'height': '84'}) }}</h2>
 {% endblock %}
 
-{% block site_css %}{% endblock %}
```
Comment (jpetto) [inline bedrock/firefox/templates/firefox/new.html]: This emptying of the `site_css` block needs to remain, otherwise modern browsers get served duplicate CSS.

This is the last thing :exclamation:

Comment (malena) [inline bedrock/firefox/templates/firefox/new.html]: @jpetto done

**Comment by jpetto** [comment]
r+!

dance
